src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -109,17 +109,21 @@ object Settings {
         ArgsSummary(sstate, args, errors :+ msg)
       def missingArg =
         fail(s"missing argument for option $name", args)
-      def doSet(argRest: String) = ((implicitly[ClassTag[T]], args): @unchecked) match {
+      def doSet(argRest: String = "") = ((implicitly[ClassTag[T]], args): @unchecked) match {
         case (BooleanTag, _) =>
           update(true, args)
         case (ListTag, _) =>
           if (argRest.isEmpty) missingArg
           else update((argRest split ",").toList, args)
-        case (StringTag, arg2 :: args2) =>
-          if (choices.nonEmpty && !(choices contains arg2))
-            fail(s"$arg2 is not a valid choice for $name", args2)
-          else
-            update(arg2, args2)
+        case (StringTag, _) =>
+          if (argRest.isEmpty && args.isEmpty) missingArg
+          else {
+            val (arg, tail) = if (argRest.isEmpty) (args.head, args.tail) else (argRest, args)
```

Comment:
We should simplify here and not accept a following argument. I suspect then the whole thing will be shorter (which is what we want).